# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/StringType-inl.h
# Commit: ae6126131b6d
# Full Hash: ae6126131b6dfa18981f25c6986200c14e63e5f2
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2024-04-11 04:26:26
# Regressor Bug: 1881995
# File Overlap Count: 1
# Description:
#   Bug 1881995 - Implement ForwardedAtoms and create them during atomization r=iain,sfink
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D202690
# ==============================================================================

diff -r 4e5967329fdd -r ae6126131b6d js/src/vm/StringType-inl.h
--- a/js/src/vm/StringType-inl.h	Wed Apr 10 15:47:15 2024 +0000
+++ b/js/src/vm/StringType-inl.h	Wed Apr 10 16:32:23 2024 +0000
@@ -376,6 +376,7 @@
     setLengthAndFlags(length, INIT_DEPENDENT_FLAGS);
     d.s.u2.nonInlineCharsTwoByte = base->twoByteChars(nogc) + start;
   }
+  base->setDependedOn();
   d.s.u3.base = base;
   if (isTenured() && !base->isTenured()) {
     base->storeBuffer()->putWholeCell(this);